Plugins/Youbora/YouboraAdnalyzerManager.swift

```diff
@@ -0,0 +1,201 @@
+    }
+    
+    override func stopMonitoring() {
+        if let messageBus = self.messageBus {
+            self.unregisterAdEvents(fromMessageBus: messageBus)
+        }
+        super.stopMonitoring()
+    }
+}
+
+/************************************************************/
+// MARK: - Youbora Info Methods
+/************************************************************/
+
+extension YouboraAdnalyzerManager {
+    
+    override func getAdPlayhead() -> NSNumber! {
+        if let adPlayhead = self.adPlayhead, adPlayhead > 0 {
+            return NSNumber(value: adPlayhead)
+        } else {
+            return super.getAdPlayhead()
+        }
+    }
+    
+    override func getAdPosition() -> String! {
+        if let adInfo = self.adInfo {
+            switch adInfo.positionType {
+            case .preRoll: return "pre"
+            case .midRoll: return "mid"
+            case .postRoll: return "post"
+            }
+        } else {
+            return super.getAdPosition()
+        }
+    }
+    
+    override func getAdTitle() -> String! {
+        return adInfo?.title ?? super.getAdTitle()
+    }
+    
+    override func getAdDuration() -> NSNumber! {
+        if let adInfo = self.adInfo {
+            return NSNumber(value: adInfo.duration)
+        } else {
+            return super.getAdDuration()
+        }
+    }
+    
+    override func getAdResource() -> String! {
+        return lastReportedResource ?? super.getAdResource()
+    }
+    
+    override func getAdPlayerVersion() -> String! {
+        return PlayKitManager.clientTag
+    }
+}
+
+/************************************************************/
+// MARK: - Events Handling
+/************************************************************/
+
+extension YouboraAdnalyzerManager {
+    
+    private var adEventsToRegister: [AdEvent.Type] {
+        return [
+            AdEvent.adLoaded,
+            AdEvent.adStarted,
+            AdEvent.adComplete,
+            AdEvent.adResumed,
+            AdEvent.adPaused,
+            AdEvent.adDidProgressToTime,
+            AdEvent.adSkipped,
+            AdEvent.adStartedBuffering,
+            AdEvent.adPlaybackReady,
+            AdEvent.adsRequested
+        ]
+    }
+    
+    fileprivate func registerAdEvents(onMessageBus messageBus: MessageBus) {
+        PKLog.debug("register ad events")
+        
+        self.adEventsToRegister.forEach { event in
+            PKLog.debug("\(String(describing: type(of: self))) will register event: \(event.self)")
+            
+            switch event {
+            case let e where e.self == AdEvent.adLoaded:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    // update ad info with the new loaded event
+                    self?.adInfo = event.adInfo
+                    // if ad is preroll make sure to call /start event before /adStart
+                    if let positionType = event.adInfo?.positionType, positionType == .preRoll {
```

Comment:
@OrenMe, thats why I prefer to have lifecycle events to count on and not play event as I spoke with you about asset loading for example.